### PR TITLE
Feature/deprecate wealth2

### PR DESCRIPTION
**Pull request by n-hutton**
This PR removes the ability for transactions to arbitrarily create wealth. Nodes must be started with a genesis file that specifies wealth at one or more addresses. For the end to end tests, this is a 'benefactor account', which is transferred from, where create wealth was previously used.

### CODE REVIEW

File: libs/ledger/src/fees/fee_manager.cpp [hunk 1]
```diff
@@ -92,14 +84,12 @@ bool FeeManager::CalculateChargeAndValidate(TransactionDetails &             tx,
...
-  if (!tx.is_create_wealth)
-  {
-    result.charge += scaled_charge;
-  }
+  result.charge += scaled_charge;
 
   // determine if the chain code ran out of charge
   if (result.charge > tx.charge_limit)
   {
+    result.charge = scaled_charge;
```
Comment (ejfitzgerald) [inline libs/ledger/src/fees/fee_manager.cpp]: This is not needed

File: libs/ledger/include/ledger/protocols/main_chain_rpc_service.hpp [hunk 2]
```diff
@@ -93,6 +93,8 @@ class MainChainRpcService : public muddle::rpc::Server,
   MainChainRpcService(MainChainRpcService &&)      = delete;
   ~MainChainRpcService() override                  = default;
 
+  void Start();
```
Comment (ejfitzgerald) [inline libs/ledger/include/ledger/protocols/main_chain_rpc_service.hpp]: Why do we need this again?

File: scripts/end_to_end_test/end_to_end_test.yaml [hunk 3]
```diff
@@ -244,7 +244,7 @@ steps:
 
 ---
 test_description: >
-  Test synergetic contract fees
+  DISABLED - Test synergetic contract fees
```
Comment (ejfitzgerald) [inline scripts/end_to_end_test/end_to_end_test.yaml:247]: How come disabled?

File: scripts/end_to_end_test/end_to_end_test.yaml [hunk 4]
```diff
@@ -267,7 +267,7 @@ steps:
 
 ---
 test_description: >
-  Test synergetic contract fees in case of no funds
+  DISABLED - Test synergetic contract fees in case of no funds
```
Comment (ejfitzgerald) [inline scripts/end_to_end_test/end_to_end_test.yaml:270]: How come disabled?

**Review by ejfitzgerald — CHANGES_REQUESTED** [review]

File: scripts/end_to_end_test/end_to_end_test.yaml (hunk 3 above)
Comment (n-hutton) [reply scripts/end_to_end_test/end_to_end_test.yaml:247]: Ran out of time, I mentioned I would ask atilla to fix them (rely on wealth creation)

File: scripts/end_to_end_test/end_to_end_test.yaml (hunk 4 above)
Comment (n-hutton) [reply scripts/end_to_end_test/end_to_end_test.yaml:270]: as above

File: libs/ledger/include/ledger/protocols/main_chain_rpc_service.hpp (hunk 2 above)
Comment (n-hutton) [reply libs/ledger/include/ledger/protocols/main_chain_rpc_service.hpp]: we do not.

**Review by ejfitzgerald — APPROVED** [review]

**Pull request merged**